server/server.go

```diff
@@ -517,16 +522,32 @@ func (server HTTPServer) fetchAggregatorReport(
 	return aggregatorResponse, true
 }
 
+func (server HTTPServer) getOSDFlag(request *http.Request) bool {
+	OSDEligible := request.URL.Query().Get("osd_eligible")
+	if len(OSDEligible) == 0 {
+		return false
+	}
+	OSDEligibleParsed, err := strconv.ParseBool(OSDEligible)
+	if err != nil {
+		log.Err(err).Msg("Got error while parsing `osd_eligible` value")
+		return false
+	}
+	return OSDEligibleParsed
+}
+
 func (server HTTPServer) reportEndpoint(writer http.ResponseWriter, request *http.Request) {
 	aggregatorResponse, successful := server.fetchAggregatorReport(writer, request)
 	if !successful {
 		return
 	}
 
 	var rules []proxy_types.RuleWithContentResponse
+	hasNoErrors := true
 
 	for _, aggregatorRule := range aggregatorResponse.Report {
-		rule, successful := server.fetchRuleContent(aggregatorRule)
+		rule, successful, isError := server.fetchRuleContent(aggregatorRule, server.getOSDFlag(request))
```

Comment:
what's the difference between successful and isError?

Reply:
`successful` indicates if rule content was fetched (it could be filtered because of OSD flag). `isError` indicate if there appear some error, for example during parsing.

Reply:
I'd prefer to use `error` instead, WDYT?